pure.zsh

```diff
@@ -324,6 +324,13 @@ prompt_pure_async_git_fetch() {
 	# Set SSH `BachMode` to disable all interactive SSH password prompting.
 	export GIT_SSH_COMMAND="${GIT_SSH_COMMAND:-"ssh"} -o BatchMode=yes"
 
+	# If gpg-agent is set to handle SSH keys for `git fetch`, make
+	# sure it uses this zpty instead of corrupting the parent TTY.
+	export GPG_TTY=$TTY
+
+	# Tell the child process to not expect any input from this zpty.
+	exec /dev/null > $TTY
```

Comment:
This broke git fetch for me. I've never seen this syntax for closing stdin, feels like it would close both stdin and stdout/break the TTY? Looks to me like this pretty much stop execution in it's tracks.

If we want to close just stdin, that'd be best accomplished by:

```sh
exec <& -
```

Should we revert the entire PR or just remove this `exec` line @sindresorhus? I'm not sure if my above suggestion will fix the issue that was attempted to fix, and don't have the facility to test it.

Reply:
I removed the `exec` part: https://github.com/sindresorhus/pure/releases/tag/v1.20.3

// @sheeviZi

Reply:
Fixed in #647 with an alternative solution. Tested locally and made sure that `pinentry-ncurses` closes and while git fetch stays unaffected.